controllers/policyendpoints_controller.go

```diff
@@ -418,11 +418,44 @@ func (r *PolicyEndpointsReconciler) updateeBPFMaps(ctx context.Context, podIdent
 	return nil
 }
 
+func (r *PolicyEndpointsReconciler) deriveTargetPodsForParentNP(ctx context.Context,
+	policyEndpoint *policyk8sawsv1.PolicyEndpoint) ([]types.NamespacedName, map[string]bool, []types.NamespacedName) {
+	var targetPods, podsToBeCleanedUp []types.NamespacedName
+	podIdentifiers := make(map[string]bool)
+	currentPE := &policyk8sawsv1.PolicyEndpoint{}
+
+	r.log.Info("Parent NP resource:", "Name: ", policyEndpoint.Spec.PolicyRef.Name)
```

Comment:
Do we need to maybe add a `nil` check? Even in the caller reconcildePolicyEndpoint we don't have the check.

Reply:
Synced up offline and reconcile ensures the object is initialized.